### PR TITLE
insights: add dashboard to worker for insights queue and queue processor

### PR DESCRIPTION
**Pull request by coury-clark**
Adds a `Worker` dashboard for the insights queue processor and queue.

### CODE REVIEW

File: enterprise/internal/insights/background/background.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func GetBackgroundJobs(ctx context.Context, mainAppDB *sql.DB, insightsDB *sql.D
 		Tracer:     &trace.Tracer{Tracer: opentracing.GlobalTracer()},
 		Registerer: prometheus.DefaultRegisterer,
 	}
-	queryRunnerWorkerMetrics, queryRunnerResetterMetrics := newWorkerMetrics(observationContext, "query_runner_worker")
+	queryRunnerWorkerMetrics, queryRunnerResetterMetrics := newWorkerMetrics(observationContext, "insights_search_queue_processor")
```
Comment (coury-clark) [inline enterprise/internal/insights/background/background.go:43]: This name didn't obey the formatting for the `workerutil` dashboards

File: monitoring/definitions/shared/codeinsights.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+var namespace string = "codeinsights"
+
+// codeInsights provides `CodeInsights` implementations.
+type codeInsights struct{}
+
+// src_insights_search_queue_total
+// src_insights_search_queue_processor_total
+func (codeInsights) NewInsightsQueryRunnerQueueGroup(containerName string) monitoring.Group {
+	return Queue.NewGroup(containerName, monitoring.ObservableOwnerCodeIntel, QueueSizeGroupOptions{
```
Comment (coury-clark) [inline monitoring/definitions/shared/codeinsights.go:15]:
```suggestion
	return Queue.NewGroup(containerName, monitoring.ObservableOwnerCodeInsights, QueueSizeGroupOptions{
```

File: monitoring/definitions/shared/codeinsights.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		`),
+	})
+}
+
+// src_insights_search_queue_processor_total
+// src_insights_search_queue_processor_duration_seconds_bucket
+// src_insights_search_queue_processor_errors_total
+// src_insights_search_queue_processor_handlers
+func (codeInsights) NewInsightsQueryRunnerWorkerGroup(containerName string) monitoring.Group {
+	return Workerutil.NewGroup(containerName, monitoring.ObservableOwnerCodeIntel, WorkerutilGroupOptions{
```
Comment (coury-clark) [inline monitoring/definitions/shared/codeinsights.go:43]:
```suggestion
	return Workerutil.NewGroup(containerName, monitoring.ObservableOwnerCodeInsights, WorkerutilGroupOptions{
```

**Comment by efritz** [comment]
Are you able to get visible metrics from the local monitoring stack?

**Comment by coury-clark** [comment]
> Are you able to get visible metrics from the local monitoring stack?

Working on verifying it now

**Comment by coury-clark** [comment]
image

**Review by slimsag — APPROVED** [review]

**Pull request merged**